### ISSUE TITLE
Block comment higlighting (with a "|")

### ISSUE THREAD

**Issue by alenkis** [purescript-emacs/purescript-mode#16]
Hi! 

I've noticed that comments like this

```purescript
--| ...
foo x = y
```

fail to properly hightlight as comments, and what I see is this:
<img width="870" alt="Screenshot 2023-11-17 at 5 17 59 PM" src="https://github.com/purescript-emacs/purescript-mode/assets/16632804/d5dc967e-4b1d-4270-beba-3cc25b5d882f">

I've tried to fix this on my end by adding

```elisp
(add-hook 'purescript-mode-hook
          (lambda ()
            (font-lock-add-keywords nil
                                    '(("--|.*" . font-lock-comment-face)))))
```

But this doesn't fully solve the problem, as ocassionally keywords appear in the comment line, which somehow breaks the comment highlighting 🤷‍♂️ 

Is this supported? Any advice on how to fix it? 
Thanks!

**Comment by kritzcreek** [comment]
> Is this supported? Any advice on how to fix it?

Sorry unfortunately I have no clue :D

My hope is that at some point we can switch over to tree-sitter based highlighting. If you do manage to figure it out, I'd be happy to accept a PR

**Comment by Hi-Angel** [comment]
Haven't looked at it, but I hope it's a bit less of a problem for the latest master, because some keywords were made top-level only, and also `default` was removed *(as it isn't a thing in PS)*, which might be a frequent word in comments as well.

**Comment by Hi-Angel** [comment]
@alenkis soo… I just looked at it and I figure I am not completely sure the syntax you're using is correct… Well, there is a bug, but I don't think it's the one we may be thinking there is…

The problem is triggered by absence of space between the dashes and the pipe, like `--|` instead of `-- |`. Whoever wrote the syntax highlight left a comment there should be a space.

I went to see what other editors have, and…

![Image](https://github.com/user-attachments/assets/4374c329-b7f5-4a53-93ff-9f31f5c8c368)

well, Visual Studio Code has exactly the same bug.

[Official documentation](https://github.com/purescript/documentation/blob/master/language/Syntax.md) also has a space, but it makes no claims on whether it's mandatory, so 🤷‍♂️ It does refer to some psc-docs though, whose behavior could be the reference we're interested in; but [purescript devel group has no such package](https://github.com/purescript?q=psc&type=all&language=&sort=).

To sum up: it is unclear if the "no-space" `--|` should be considered a plain comment or a documentation comment.

I'm going to CC @megamaddu for good measure as the author of the comments on screenshot [in this file](https://github.com/purescript-react/purescript-react-basic-hooks/blob/main/src/React/Basic/Hooks.purs), since we have at least two major code editors where they're rendered incorrectly 😊

**Comment by Hi-Angel** [comment]
Or maybe @kritzcreek knows, since he's in the thread as well.

**Comment by Hi-Angel** [comment]
At the same time, looking at [pursuit documentation for the problematic file](https://pursuit.purescript.org/packages/purescript-react-basic-hooks/8.2.0/docs/React.Basic.Hooks), it renders just fine. Omg, why is everything so complicated…
